controllers/client.go

```diff
@@ -0,0 +1,9 @@
+package controllers
+
+import "sigs.k8s.io/controller-runtime/pkg/client"
+
+//go:generate mockgen -source=client.go -package=controllers -destination=mock_client.go
+
+type Client interface {
+	client.Client
+}
```

Comment:
This is the only way I managed to generate *only* the mock for the `client.Client` interface because if we use the original source file `sigs.k8s.io/controller-runtime@v0.11.2/pkg/client/interfaces.go` it generates the code for 8 interfaces.

Reply:
Just to make sure I understand, how do you generate the mock by using `sigs.k8s.io/controller-runtime@v0.11.2/pkg/client/interfaces.go`?

Also I would suggest moving this file to `pkg/client/client.go` but as all our code is currently in `controller/` for some reason maybe it is better to leave it here and have another task that moves all the packages to `pkg/` across the whole code.

Toughts @yevgeny-shnaidman?

Reply:
I think we can make all the changes in one PR, lets just rename the PR and Jira accordingly

Reply:
done :heavy_check_mark: